service/controller/resource/monitoring/ingress/resource.go

```diff
@@ -114,6 +118,7 @@ func toIngress(v interface{}, config Config) (metav1.Object, error) {
 		},
 		ObjectMeta: objectMeta,
 		Spec: networkingv1beta1.IngressSpec{
+			IngressClassName: &ingressClassName,
```

Comment:
How does this works on 1.18 k8s?

Reply:
I'm trying to deploy it on 1.20 first :D

Reply:
Yes this is not working

Reply:
I wanted to see if we could use chartutil to get the capabilities but it looks like we cannot support v1beta1 and v1 easily enough :(

Reply:
We should know the kubernetes version we are working against, but then we need to maintain both
`v1beta1.Ingress` and `v1.Ingress` objects in Go also :face_exhaling:

Reply:
I think maintaining the 2 would not be hard tbh but that means we need to move out of the generic resource and tests :( I'm not sure I want to go that way today :D

Reply:
Maybe we can find faith together tomorrow and pair on this

Reply:
Tested on gorgoth with 1.18